contrib/language/filters/http/source/config.cc

```diff
@@ -0,0 +1,69 @@
+#include "contrib/language/filters/http/source/config.h"
+
+#include "envoy/common/exception.h"
+#include "envoy/registry/registry.h"
+
+#include "contrib/language/filters/http/source/language_filter.h"
+#include "unicode/localematcher.h"
+#include "unicode/utypes.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Language {
+
+Http::FilterFactoryCb LanguageFilterFactory::createFilterFactoryFromProtoTyped(
+    const envoy::extensions::filters::http::language::v3alpha::Language& proto_config,
+    const std::string& stats_prefix, Server::Configuration::FactoryContext& context) {
+  const auto default_locale = icu::Locale(proto_config.default_language().data());
+
+  if (default_locale.isBogus()) {
+    throw EnvoyException(fmt::format("Failed to create icu::Locale from default_language: {}",
+                                     proto_config.default_language().data()));
+  }
+
+  std::vector<icu::Locale> supported_languages;
```

Comment:
should this be a set?

Reply:
Replaced the implementation with a std::set of icu::Locale and added a custom comparator.

https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/classicu_1_1Locale.html#a9423f4eb4b671eec3d98c26ec923351e